File: Javascript/breadth-first-search.js
// Define the node constructor function

function node(data) {
  this.data = data;
  this.left = null;
  this.right = null;
}

// Define the breadthFirstSearch function

function breadthFirstSearch(node, value) {
  var queue = [node];
  while (queue.length > 0) {
    var currentNode = queue.shift();
    if (currentNode.data === value) {
      return currentNode;
    } else {
      if (currentNode.left) {
        queue.push(currentNode.left);
      }
      if (currentNode.right) {
        queue.push(currentNode.right);
      }
    }
  }
  return "Value not found";
}

// Create a tree
var headNode = new node(5);
headNode.left = new node(4);
headNode.right = new node(24);
headNode.left.left = new node(47);
headNode.left.right = new node(64);
headNode.right.left = new node(2);
headNode.right.right = new node(3);

// Run breadthFirstSearch function
console.log(breadthFirstSearch(headNode, 24));

File: Python/breadth-first-search.py
# Define a node class that will be used to build trees

class Node(object):
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

    def __repr__(self):
        return "Data: %s, Left: %s, Right: %s" % (self.data, self.left, self.right)

# Write a breadth-first search function

def breadth_first_search(node, value, queue = []):
    if node.data == value:
        return node

    if node.left:
        queue.append(node.left)
    if node.right:
        queue.append(node.right)

    if len(queue) > 0:
        next_item = queue.pop(0)
        return breadth_first_search(next_item, value, queue)
    else:
        return "Value not found"

# Create a tree

head_node = Node(5)
head_node.left = Node(4)
head_node.right = Node(3)
head_node.left.left = Node(10)
head_node.left.right = Node(12)
head_node.right.left = Node(2)
head_node.right.right = Node(8)

# Call breadth-first search function on tree

print breadth_first_search(head_node, 2)
